djvm/src/test/java/net/corda/djvm/execution/JavaObjectTest.java

```diff
@@ -0,0 +1,244 @@
+            } catch (Exception e) {
+                throw new RuntimeException(e.getMessage(), e);
+            }
+        }
+    }
+
+    @Test
+    void testNotifyReferenceIsForbidden() {
+        sandbox(ctx -> {
+            try {
+                TypedTaskFactory taskFactory = ctx.getClassLoader().createTypedTaskFactory();
+                RuleViolationError ex = assertThrows(RuleViolationError.class,
+                    () -> WithJava.run(taskFactory, ObjectNotify.class, null));
+                assertThat(ex)
+                    .hasMessage("Disallowed reference to API; java.lang.Object.notify()");
+            } catch (Exception e) {
+                fail(e);
+            }
+        });
+    }
+
+    public static class ObjectNotify implements Function<Object, Object> {
+        @Override
+        public Object apply(Object input) {
+            Consumer<? super Object> notifier = Object::notify;
+            notifier.accept(input);
+            return null;
+        }
+    }
+
+    @Test
+    void testNotifyAllReferenceIsForbidden() {
+        sandbox(ctx -> {
+            try {
+                TypedTaskFactory taskFactory = ctx.getClassLoader().createTypedTaskFactory();
+                RuleViolationError ex = assertThrows(RuleViolationError.class,
+                    () -> WithJava.run(taskFactory, ObjectNotifyAll.class, null));
+                assertThat(ex)
+                    .hasMessage("Disallowed reference to API; java.lang.Object.notifyAll()");
+            } catch (Exception e) {
+                fail(e);
+            }
+        });
+    }
+
+    public static class ObjectNotifyAll implements Function<Object, Object> {
+        @Override
+        public Object apply(Object input) {
+            Consumer<? super Object> notifier = Object::notifyAll;
+            notifier.accept(input);
+            return null;
+        }
+    }
+
+    @Test
+    void testHashCodeReference() {
+        sandbox(ctx -> {
+            try {
+                TypedTaskFactory taskFactory = ctx.getClassLoader().createTypedTaskFactory();
+                Object result = WithJava.run(taskFactory, GetHashCode.class, new Object());
+                assertThat(result).isEqualTo(0xfed_c0de + 1);
+            } catch (Exception e) {
+                fail(e);
+            }
+        });
+    }
+
+    public static class GetHashCode implements Function<Object, Integer> {
+        @Override
+        public Integer apply(Object input) {
+            ToIntFunction<? super Object> hashCode = Object::hashCode;
+            return hashCode.applyAsInt(input);
+        }
+    }
+
+    @Test
+    void testStringReference() {
+        UUID uuid = UUID.randomUUID();
+        sandbox(ctx -> {
+            try {
+                TypedTaskFactory taskFactory = ctx.getClassLoader().createTypedTaskFactory();
+                Object result = WithJava.run(taskFactory, GetString.class, uuid);
+                assertThat(result).isEqualTo(uuid.toString());
```

Comment:
Should we also be asserting it is `isNotSameAs`?

Reply:
Ah, you mean checking that the `uuid` object is _really_ from the sandbox? Yes, we could do that :+1:.